### PR TITLE
render in terminal via blockish

### PR DESCRIPTION
**Pull request by yazgoo**
- adds terminal rendering w/ blockish (`-t` flag)
- handles input w/ crossterm_input

video demos:

zelda: https://www.youtube.com/watch?v=ou_4WSuYRHk
pokemon yellow: https://youtu.be/XcrkZnIS1lM

### CODE REVIEW

**Comment by mohanson** [comment]
I found that you have removed the WIP tag, but it still seems to be doing some work. If the PR is ready, please @ me, I will test and check it.

Thank you for your work.

**Comment by yazgoo** [comment]
@mohanson yep, the PR is ready :)

**Comment by mohanson** [comment]
@yazgoo 

1. Please fix their build warnings:
```no-highlight
warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> src\main.rs:128:26
    |
128 |         None => {let _ = execute!(stdout(),terminal::EnterAlternateScreen);},
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> src\main.rs:161:25
    |
161 |                 let _ = execute!(stdout(),cursor::MoveTo(0,0));
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> src\main.rs:227:26
    |
227 |         None => {let _ = execute!(stdout(),terminal::LeaveAlternateScreen);},
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 3 warnings emitted
```

2. This UI does not work well on windows, you need to fix it, or simply disable the use of terminal mode on windows, both ok.

image

File: src/main.rs [hunk 1]
```diff
@@ -40,10 +52,18 @@ fn main() {
         8 => minifb::Scale::X8,
         _ => panic!("Supported scale: 1, 2, 4 or 8"),
     };
-    let mut window =
-        minifb::Window::new(format!("Gameboy - {}", rom_name).as_str(), SCREEN_W, SCREEN_H, option).unwrap();
+    let mut window_opt = None;
+    if !c_terminal {
```
Comment (mohanson) [inline src/main.rs:56]: The judgment condition here should only be established when the user enters -t flag, the correct one should be `if c_terminal`, not `if !c_terminal`.

**Comment by yazgoo** [comment]
Thanks !

**Pull request merged**